File: CastIt.Domain/Dtos/Responses/PlayerStatusResponseDto.cs
namespace CastIt.Domain.Dtos.Responses
{
    public class PlayerStatusResponseDto
    {
        public string Mrl { get; set; }

        public bool IsPlaying { get; set; }
        public bool IsPaused { get; set; }
        public bool IsPlayingOrPaused { get; set; }

        public double CurrentMediaDuration { get; set; }
        public double ElapsedSeconds { get; set; }
        public double PlayedPercentage { get; set; }

        public double VolumeLevel { get; set; }
        public bool IsMuted { get; set; }
    }
}

File: CastIt.Client/ios/CastIt Watch App/Models/Responses/PlayerStatusResponseDto.swift
import Foundation

// Mirrors ClientApp IPlayerStatusResponseDto
struct PlayerStatusResponseDto: Codable {
    let mrl: String?
    let isPlaying: Bool
    let isPaused: Bool
    let isPlayingOrPaused: Bool

    let currentMediaDuration: Double
    let elapsedSeconds: Double
    let playedPercentage: Double

    let volumeLevel: Double
    let isMuted: Bool
}
